src/main/java/org/embulk/filter/ColumnFilterPlugin.java

```diff
@@ -250,12 +287,22 @@ public PageOutput open(final TaskSource taskSource, final Schema inputSchema,
             outputInputColumnMap.put(outputColumn, inputColumn); // NOTE: inputColumn would be null
         }
 
-        // Map outputColumn => default value if present
+        // Map outputColumn => src and default value if present
+        final HashMap<Column, String> outputSrcMap = new HashMap<Column, String>();
         final HashMap<Column, Object> outputDefaultMap = new HashMap<Column, Object>();
+
         for (Column outputColumn: outputSchema.getColumns()) {
             String name = outputColumn.getName();
             Type   type = outputColumn.getType();
 
+            String src_name = getSrc(name, task.getColumns());
+            if (src_name == null){
+                src_name = getSrc(name, task.getAddColumns());
+            }
+            if (src_name != null) {
+                outputSrcMap.put(outputColumn, src_name);
+            }
+
```

Comment:
Can't you get inputColumn directly from src_column_name, and put into `outputInputColumnMap`?